c7n/filters/vpc.py

```diff
@@ -221,6 +237,15 @@ def process_resource(self, r, resource_sgs, resource_subnets, key):
                     'reason': 'ResourceLocationMismatch',
                     'resource': r_value,
                     'subnet': subnet_values})
+            if 'security-group' in self.compare and resource_sgs:
+                temp_sgs = {sg_id: sg_value
+                            for sg_id, sg_value in sg_values.items()
+                            if sg_value != r_value}
+                if len(temp_sgs) > 0:
```

Comment:
just use an `if mapping`, the length check is a bit superflous given natural boolean evaluation of dicts.